types/plugin.ts

```diff
@@ -37,3 +43,302 @@ export const Plugins: Record<PluginID, Plugin> = {
+): OpenAIFunction {
+  const {
+    operationId: id,
+    summary,
+    parameters,
+    requestBody,
+  } = pluginApiOperation;
+
+  // Build the properties object for OpenAIFunction
+  const properties: { [key: string]: {} } = {};
+  if (parameters) {
+    for (const parameter of parameters) {
+      properties[parameter.name] = parameter.schema;
+    }
+  }
+
+  // Build the required array for OpenAIFunction
+  const required =
+    parameters
+      ?.filter((parameter) => parameter.required)
+      .map((parameter) => parameter.name) || [];
+
+  if (requestBody) {
+    for (const key in requestBody.content) {
+      for (const propertyKey in requestBody.content[key].schema.properties) {
+        properties[propertyKey] = {
+          type: requestBody.content[key].schema.properties[propertyKey].type,
+          description:
+            requestBody.content[key].schema.properties[propertyKey]
+              .description || '',
+        };
+        if (requestBody.content[key].schema.properties[propertyKey].required) {
+          required.push(propertyKey);
+        }
+      }
+    }
+  }
+
+  // Create OpenAIFunction object
+  const openAIFunction: OpenAIFunction = {
+    name: id,
+    description: summary || '',
+    parameters: {
+      type: 'object',
+      properties: properties,
+      required: required,
+    },
+  };
+
+  return openAIFunction;
+}
+
+export const runPluginApiOperation = async (
+  operation: PluginApiOperation,
+  args: string,
+) => {
+  let query: { [key: string]: string } = {};
+  if (operation.parameters) {
+    for (const parameter of operation.parameters) {
+      if (parameter.in === 'path' && args.includes(parameter.name)) {
+        const newPath = JSON.parse(args)[parameter.name];
+        operation.apiPath = operation.apiPath.replace(
+          `{${parameter.name}}`,
+          newPath,
+        );
+      }
+      if (parameter.in === 'query' && args.includes(parameter.name)) {
+        const newQuery = JSON.parse(args)[parameter.name];
+        query[parameter.name] = newQuery;
+      }
+    }
+  }
+
+  let body: any = {};
+  if (operation.requestBody) {
+    for (const key in operation.requestBody.content) {
+      for (const propertyKey in operation.requestBody.content[key].schema
+        .properties) {
+        if (args.includes(propertyKey)) {
+          body[propertyKey] = JSON.parse(args)[propertyKey];
+        }
+      }
+    }
+    body = JSON.stringify(body);
+  } else {
+    body = undefined;
+  }
+
+  let url = operation.serverUrl + operation.apiPath;
+  if (query) {
+    url = url + '?' + new URLSearchParams(query).toString();
+  }
+
+  const response = await fetch(url, {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    method: operation.method,
+    body: body,
+  });
+
+  if (operation.responses) {
+    for (const key in operation.responses) {
+      if (response.status === parseInt(key)) {
+        if (operation.responses[key].content) {
```

Comment:
Check for `operation.responses[key].content` breaks some of the plugins. For example, "VoxScript" (`https://voxscript.awt.icu`) doesn't have the `content` property in its responses:
```js
{
    operationId: 'GetYoutubeVideoData',
    nameForModel: 'VoxScript',
    serverUrl: 'https://voxscript.awt.icu',
    apiPath: '/GetYoutubeVideoData',
    method: 'post',
    summary: 'Gets info on a YouTube video by channel URL, channel ID, Video ID, Video URL.',
    parameters: [{
        name: 'input',
        in: 'query',
        description: 'Channel URL, channel ID, Video ID, Video URL',
        required: undefined,
        schema: {
            type: 'string'
        }
    }],
    responses: {
        200: {
            description: 'Success'
        }
    }
}
```

I would suggest simplifying this code:
```js
  if (operation.responses) {
    if (response.status === 200) {
      return await response.json();
    } else {
      return response.statusText;
    }
  }
```